File: users/src/main/java/com/deck/users/services/UserService.java
package com.deck.users.services;

import com.deck.users.dto.PublicUserDTO;
import com.deck.users.dto.UserDTO;
import com.deck.users.models.User;
import org.springframework.data.domain.Pageable;
import org.springframework.security.provisioning.UserDetailsManager;

import java.util.List;
import java.util.Optional;

public interface UserService extends UserDetailsManager {
    List<User> findAll();
    Optional<User> findById(Long id);
    Optional<User> findByEmail(String email);
    List<User> getUsersByIds(Iterable<Long> ids);
    List<PublicUserDTO> searchUsersByUsername(String username, Pageable pageable);
    User save(User user);
    UserDTO save(UserDTO user);
    UserDTO updateUser(UserDTO user, Long userId);
    void delete(Long id);
}

File: frontend/src/services/user/UserService.ts
import User from "../../models/user/User";
import { GenericClient } from "../genericClient/genericClient";

const instance = new GenericClient("/users");

export const getUserDetails = (userId: number | string) => {
  return instance.get<User>(`/${userId}`);
};

export const searchUsersByUsername = (username: string) => {
  return instance.get<User[]>(`/search-by-username?username=${username}`);
};
